src/hal/user_comps/sendkeys.c

```diff
@@ -107,8 +107,12 @@ int init(int argc, char* argv[]){
                     v1 = realloc(codes, (index + 1) * sizeof(int));
                     v2 = realloc(pins, (index + 1) * sizeof(int));
                     if (!v1 || !v2) {
-                        free(codes);
-                        free(pins);
+                        if (!v1) {
+                            free(codes);
```

Comment:
A good defensive coding strategy is to assign NULL to any pointer that has been freed.  Perhaps a good idea to do here too?  I know the variables are local and there is a return a few lines below.